e2e/txpool_test.go

```diff
@@ -408,44 +408,6 @@ func TestTxPool_StressAddition(t *testing.T) {
 	}
 }
 
-func TestInvalidTransactionRecover(t *testing.T) {
```

Comment:
Please link the unit test(s) that cover this functionality in the new TxPool implementation 🙏

Reply:
This PR will discard every transaction if they are sent with an invalid gas limit, no transaction will pass validationTx with an invalid gas limit. 
There is no way to write unit/e2e test which will test edge-case when transaction is valid when arrives in txPool (latest block is valid) but new block is invalid (while building)

Reply:
Why not test adding the transaction?

You can have a few test cases for valid gas limits, and some for invalid

Reply:
In the txpool_test.go we have a unit test that covers adding error tx (TestAddTxErrors). It covers all validation rules and because I have added a new validation rule I had to add a new test case for this rule. 

https://github.com/0xPolygon/polygon-edge/pull/349/files#diff-c6d10affe0540289b09ab01da147d827eb2c3aa0a899b5e5d5c43810821422cbR113

In my opinion, because this is mock with defined values this one test covers the validation check for ErrBlockLimitExceeded. Adding multiple test cases for valid and invalid values would just be redundant check if one value is bigger than the other.

And every other test which adds a new transaction inside this package will show that if we have a valid gas limit (tx.gasLimit < latestBlockGasLimit) will be considered valid